rpmio/rpmkeyring.c

```diff
@@ -182,14 +185,26 @@ rpmPubkey *rpmGetSubkeys(rpmPubkey mainkey, int *count)
 	/* Returned to C, can't use new */
 	subkeys = (rpmPubkey *)xmalloc(pgpsubkeysCount * sizeof(*subkeys));
 	for (i = 0; i < pgpsubkeysCount; i++)
-	    subkeys[i] = rpmPubkeyNewSubkey(pgpsubkeys[i]);
+	    subkeys[i] = rpmPubkeyNewSubkey(mainkey, pgpsubkeys[i]);
 	free(pgpsubkeys);
     }
     *count = pgpsubkeysCount;
 
     return subkeys;
 }
 
+
+int rpmPubkeyFingerprint(rpmPubkey key, uint8_t **fp, size_t *fplen)
+{
+    if (key == NULL)
+	return -1;
+    key = key->mainkey;
+    pthread_rwlock_rdlock(&key->lock);
```

Comment:
The lock should be one line earlier, above the key assignment.

Reply:
Not quite sure about that. To be fair may be I should just use an new variable for that. 
But right now we are locking the main key. The one we are actually processing. For sub keys we might need to lock that, too, if the assignment needs a lock around it. Ofc we'd need to be more careful for main keys to not lock them twice.

Reply:
OK, now that the locking is completely different I have re-done this.